### ISSUE TITLE
Resolve pylintrc.toml in pylint 3.x

### ISSUE THREAD

**Issue by matejsp** [pylint-dev/pylint#9228]
### Current problem

We are trying to migrate to toml format for pylintrc however if we use
pylintrc it does not detect toml format.

We don't want to have couple of hundrets of lines in pyproject.toml that's why we would like to preserve the external configuration.

### Desired solution

Based on this:
https://pylint.pycqa.org/en/latest/user_guide/usage/run.html

Have an option to resolve pylint.toml or pylintrc.toml in addition to pylintrc and .pylintrc.
One alternative is to have the same toml detection as for argument --rcfile.

Current workaround:
PYLINTRC=pylintrc.toml pylint -v
pylint --rcfile=pylintrc.toml -v

Both options are correctly detecting toml files.

**Comment by matejsp** [comment]
I found and option for rcfile= that from page:  https://www.codeac.io/documentation/pylint-configuration.html

I tried adding this to pylintrc
```
[tool.pylint.main]
rcfile = "pylintrc.toml"
```

Then I tried adding this to pyproject.toml:
```
[tool.pylint.main]
rcfile = "pylintrc.toml"
```

but it is not working ...

**Comment by Pierre-Sassoulas** [comment]
Hello, and thank you for opening the issue. Did you consider only porting the non default value to the ``pyproject.toml`` or do you actually have a lot of non default value in the pylintrc ? The generated pylintrc contain everything when generated but most of the time ``disable`` and ``load-plgins`` is the only thing I need. Reading multiple configuration file is a complexity I'd rather avoid to be honest.

**Comment by matejsp** [comment]
Hello, yes and it is still big (a lot of exclusions, rules, pylint plugins and configuration for plugins ...) and would like not to avoid doubling the pyproject.toml file. Even today is is hard to maintain.

From what I saw ... 
imagine you would like to move your own https://github.com/pylint-dev/pylint/blob/main/pylintrc to toml format ...
basically rename to pylintrc.toml and migrate data to toml and it should all just work 🥇 .

Basically the logic is already there. If file ends with toml it is parsed as toml:
https://github.com/pylint-dev/pylint/blob/main/pylint/config/config_file_parser.py#L111

So the only change needed is to add new file with toml ext to the list:
https://github.com/pylint-dev/pylint/blob/main/pylint/config/find_default_config_files.py#L18

```
RC_NAMES = (Path("pylintrc"), Path("pylintrc.toml"), Path(".pylintrc"), Path(".pylintrc.toml"))
```

Of coruse if you agree with the solution.

I am still a bit puzzled with rcfile option in configuration file ... What is the use case.

**Comment by Pierre-Sassoulas** [comment]
> I am still a bit puzzled with rcfile option in configuration file ... What is the use case.

We just have parity between command line option and configuration file option. For rcfile if doesn't make sense. This issue is slighlty related to #5833 imo (we need to be able to merge new config into already loaded one).
